### DISCUSSION TITLE
some directories of /OpenROAD-flow-scripts/tools are empty， even if i use —recursive to clone.Is there solution for the problem？

### DISCUSSION THREAD

**Discussion by yMichaelqh** [The-OpenROAD-Project/OpenROAD-flow-scripts#1626] [Q&A] [answered]
I had cloned /OpenROAD-flow-scripts,  but LSOracle, OpenROAD, yosys are empty. Configuration I used is Centos7.

**Comment by rovinski** [comment]
Try navigating inside `/OpenROAD-flow-scripts` and doing
```
git submodule init
git submodule update
```

**Reply by yMichaelqh** [reply]
thx，it‘s heplful

**Comment by maliberty** [comment, marked as answer]
When you run build.sh it will automatically do what @rovinski suggests (btw, git submodule update --init --recursive is simpler and recursive)

**Reply by yMichaelqh** [reply]
thx, it's helpful
